src/Extension/ValueResolvers/OptionValueResolver.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Extension\ValueResolvers;
+
+use SprykerSdk\Sdk\Core\Appplication\Exception\InvalidSettingException;
+use SprykerSdk\Sdk\Core\Appplication\ValueResolver\AbstractValueResolver;
+use SprykerSdk\SdkContracts\Entity\ContextInterface;
+use SprykerSdk\SdkContracts\ValueResolver\ConfigurableValueResolverInterface;
+
+class OptionValueResolver extends AbstractValueResolver implements ConfigurableValueResolverInterface
+{
+    /**
+     * @var string
+     */
+    protected const OPTION_TYPE = 'option';
+
+    /**
+     * @var string
+     */
+    protected const ARGUMENT_TYPE = 'argument';
+
+    /**
+     * @var string
+     */
+    protected string $type = '';
+
+    /**
+     * @var string
+     */
+    protected string $name = '';
+
+    /**
+     * @var string
+     */
+    protected string $description = '';
+
+    /**
+     * @param \SprykerSdk\SdkContracts\Entity\ContextInterface $context
+     * @param array $settingValues
+     * @param bool $optional
+     *
+     * @return mixed
+     */
+    public function getValue(ContextInterface $context, array $settingValues, bool $optional = true): mixed
+    {
+        $value = parent::getValue($context, $settingValues, true);
+
+        if (!$value) {
+            return null;
+        }
+
+        if ($this->type === static::OPTION_TYPE) {
+            return sprintf('--%s=%s', $this->name, $value);
+        }
+
+        if ($this->type === static::ARGUMENT_TYPE) {
```

Comment:
Isn't this logic already covered by the StaticValueResolver?

Reply:
Fixed